yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -125,8 +125,11 @@ private[spark] abstract class YarnSchedulerBackend(
    * This includes executors already pending or running.
    */
   override def doRequestTotalExecutors(requestedTotal: Int): Boolean = {
+
+    val nodeBlacklist: Set[String] = scheduler.blacklistTracker.nodeBlacklist()
```

Comment:
this is safe to call without a lock on the task scheduler because the nodeBlacklist is stored in an AtomicReference, which is set whenever the node blacklist changes.  We could just get a lock on the task scheduler here -- but I felt that it would just be harder to ensure correctness (not just in this change, but making sure the right lock was always held through future changes), and the only real cost is that we duplicate the set of nodes in the blacklist, which is hopefully very small.